### PR TITLE
Feature/toe prefill sponsors

### PR DESCRIPTION
**Pull request by FernandoBoza**
## Description
Fetches and fills out TOE form, sponsors chapters.

## Testing done
- [ ] Done

### CODE REVIEW

File: src/applications/toe/reducers.js [hunk 1]
```diff
@@ -24,6 +27,36 @@ export default {
...
+          personalInfoFetchInProgress: false,
+          formData: action?.response || {},
+          fetchedSponsorsComplete: true,
+          sponsors: {
+            sponsors: [
+              ...action?.response?.data?.attributes?.toeSponsors?.transferOfEntitlements.map(
+                (sponsor, index) => {
+                  return {
+                    ...sponsor,
+                    // ! TODO: CHANGE ID
```
Comment (nathanielbuck) [inline src/applications/toe/reducers.js:49]: Think you forgot to revert this change for testing.

Comment (FernandoBoza) [reply src/applications/toe/reducers.js:49]: Should that be reverted after testing? At the moment, the api's id's are all the same.

File: src/applications/toe/helpers.jsx [hunk 2]
```diff
@@ -99,29 +99,64 @@ export function prefillTransformer(pages, formData, metadata, state) {
...
     formId: state.data?.formData?.data?.id,
     claimantId: claimant.claimantId,
-    'view:userFullName': {
-      userFullName: {
-        first: claimant.firstName || undefined,
-        middle: claimant.middleName || undefined,
-        last: claimant.lastName || undefined,
+    [formFields.viewUserFullName]: {
+      [formFields.userFullName]: {
+        first: profile?.userFullName?.first || claimant?.firstName || undefined,
```
Comment (nathanielbuck) [inline src/applications/toe/helpers.jsx:141]: This logic is wrong. I just realized that you're not taking the vaProfile object (commented out above) into account. That should be the first thing we reference when prefilling.

**Review by nathanielbuck — CHANGES_REQUESTED** [review]

File: src/applications/toe/helpers.jsx [hunk 3]
```diff
@@ -99,29 +99,72 @@ export function prefillTransformer(pages, formData, metadata, state) {
...
     formId: state.data?.formData?.data?.id,
     claimantId: claimant.claimantId,
-    'view:userFullName': {
-      userFullName: {
-        first: claimant.firstName || undefined,
-        middle: claimant.middleName || undefined,
-        last: claimant.lastName || undefined,
+    [formFields.viewUserFullName]: {
+      [formFields.userFullName]: {
+        first:
```
Comment (nathanielbuck) [inline src/applications/toe/helpers.jsx:141]: When I implemented this for MEB, I treated name as a block. So, just in case names are different, they wouldn't be mixed and matched.

If the VA profile is {givenNames: ["John"], familyName: "Smith} and the claimant call returns {firstName: "Jane", middleName: "Louise", lastName: "Doe"}, the result with your logic would be "John Louise Smith".

This is my implementation: https://github.com/department-of-veterans-affairs/vets-website/pull/22215/files#diff-61b93329fb83366b254dccccdcb3db442c05dc3325d498e1966822db818878c8R228
